File: externals/Async-CTP/samples/VB/Windows/Phone/Concert/Finder/Concert/Finder/Converters/BooleanVisibilityConverter.vb
Imports System
Imports System.Globalization
Imports System.Windows
Imports System.Windows.Data

Namespace ConcertFinder.Converters

    ''' <summary>
    ''' Converter for boolean values.
    ''' </summary>
    Public Class BooleanVisibilityConverter
        Implements IValueConverter

        ''' <summary>
        ''' Convert a boolean value to a visibility.
        ''' </summary>
        Public Function Convert(value As Object, targetType As [Type], parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim convertedValue = Visibility.Visible
            Dim originalValue = DirectCast(value, Boolean)
            Dim isInverted = (If(TryCast(parameter, String), [String].Empty)).[Equals]("IsInverted")
            If originalValue = True Then
                convertedValue = If(Not isInverted, Visibility.Visible, Visibility.Collapsed)
            Else
                convertedValue = If(Not isInverted, Visibility.Collapsed, Visibility.Visible)
            End If
            Return convertedValue
        End Function

        Public Function ConvertBack(value As Object, targetType As [Type], parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function

    End Class

End Namespace

File: externals/Async-CTP/samples/CS/Windows/Phone/Concert/Finder/Concert/Finder/Converters/BooleanVisibilityConverter.cs
using System;
using System.Globalization;
using System.Windows;
using System.Windows.Data;

namespace ConcertFinder.Converters
{
    /// <summary>
    /// Converter for boolean values.
    /// </summary>
    public class BooleanVisibilityConverter : IValueConverter
    {
        /// <summary>
        /// Convert a boolean value to a visibility.
        /// </summary>
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            var convertedValue = Visibility.Visible;

            var originalValue = (bool)value;
            var isInverted = (parameter as string ?? String.Empty).Equals("IsInverted");
            if (originalValue == true)
            {
                convertedValue = !isInverted ? Visibility.Visible : Visibility.Collapsed;
            }
            else
            {
                convertedValue = !isInverted ? Visibility.Collapsed : Visibility.Visible;
            }

            return convertedValue;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}
